ister_gui.py

```diff
@@ -641,24 +643,215 @@ def validate_condition(self):
+        # allow time to start network unit on boot
+        time.sleep(.8)
+        super(NetworkRequirements, self).__init__()
+        static_button = urwid.Button('Set static ip configuration',
+                                     on_press=self._static_configuration)
+        reset_button = urwid.Button('Reset network to default configuration',
+                                    on_press=self._reset_network)
+        proxy_button = urwid.Button('Set installer proxy settings',
+                                    on_press=self._set_proxy)
+        static_button = urwid.AttrMap(static_button, None,
+                                      focus_map='reversed')
+        reset_button = urwid.AttrMap(reset_button, None,
+                                     focus_map='reversed')
+        proxy_button = urwid.AttrMap(proxy_button, None,
+                                     focus_map='reversed')
+        self.static_col = urwid.Columns([static_button, urwid.Divider()])
+        self.reset_col = urwid.Columns([reset_button, urwid.Divider()])
+        self.proxy_col = urwid.Columns([proxy_button, urwid.Divider()])
+        self.installer_proxy = urwid.Edit('installer proxy (https): ', '')
+        self.static_ip = None
+        self.interface = None
+        self.gateway = None
+        self.error = None
+        self.config = None
+        self.ifaceaddrs = None
+
     def handler(self, config):
-        alert_content = 'Step 1 of 5\n\n'                                     \
-                        'Network Requirements:\n'                             \
-                        '  * Wired network connection\n'                      \
-                        '  * DHCP\n'                                          \
-                        '  * Internet Access\n\n'                             \
-                        'The installer will navigate most environments with ' \
-                        'autoproxies automatically\n'
-
-        alert = Alert("Clear Linux OS for Intel Architecture Installer",
-                      alert_content)
-        alert.do_alert()
-        return alert.response
+        # make config an insance variable so we can copy proxy settings to it
+        self.config = config
+
+        # normally much of this would belong in __init__, but we want it
+        # refreshed every time the user refreshes/returns to the screen
+        self.build_ui_widgets()
+        self.build_ui()
+        self._action = self.run_ui()
+        if self._action == 'Refresh':
+            # re-build the widgets
+            return self._action
+
+        self.error = ''
+        try:
+            ipaddress.ip_address(self.static_ip.get_edit_text())
+        except Exception as err:
+            self.error = 'The configuration for "ip" is invalid {}'.format(err)
+            # an empty action signals to refresh the page
+            self._action = ''
+        if self.interface.get_edit_text() not in self.ifaceaddrs.keys():
```

Comment:
Since I haven't run this, is the user able to set configurations for interfaces that don't exist?